src/pyfreeleh/row/gsheet.py

```diff
@@ -1,412 +1,458 @@
-import time
-from typing import Any, Dict, List, Optional, Tuple
+from typing import Any, Dict, Generic, List, Type, TypeVar
 
-from pyfreeleh.base import Codec, InvalidOperationError
-from pyfreeleh.codec import BasicCodec
 from pyfreeleh.providers.google.auth.base import GoogleAuthClient
 from pyfreeleh.providers.google.sheet.base import A1CellSelector, A1Range, BatchUpdateRowsRequest
-from pyfreeleh.providers.google.sheet.wrapper import GoogleSheetWrapper
+from pyfreeleh.providers.google.sheet.wrapper import GoogleSheetSession, GoogleSheetWrapper
 from pyfreeleh.row.base import Ordering
+from pyfreeleh.row.models import Model, PrimaryKeyField
+from pyfreeleh.row.query_builder import GoogleSheetQueryBuilder
+from pyfreeleh.row.serializers import FieldColumnMapper, ModelGoogleSheetSerializer, Serializer
 
 
 class InvalidQuery(Exception):
     pass
 
 
-class QueryBuilder:
-    WHERE_TEMPLATE = "{} IS NOT NULL AND {}"
-    TS_FIELD = "_ts"
+T = TypeVar("T", bound=Model)
 
-    def __init__(self, col_mapping: Dict[str, str]):
-        self._where: Optional[Tuple[str, Tuple[Any, ...]]] = None
-        self._orderings: List[Tuple[str, Ordering]] = []
-        self._limit: int = 0
-        self._offset: int = 0
 
-        self._col_mapping = col_mapping
+def map_orderings(mapper: FieldColumnMapper, orderings: List[Ordering]) -> List[Ordering]:
+    mapped_orderings = []
+    for order in orderings:
+        col_name = mapper.to_column(order._field_name)
+        mapped_ordering = order.copy()
```

Comment:
Any reason why you want to copy first? Just for clean flow?

Reply:
To ensure it doesn't mutate the original instance.